xdsl/dialects/onnx.py

```diff
@@ -673,6 +675,82 @@ def verify_(self) -> None:
                 f"Only one value attribute must be provided, but {used_attrs} were specified"
             )
 
+    def print(self, printer: Printer):
+        if self.value:
+            printer.print_string(" ")
+            printer.print_attribute(self.value)
+        printer.print_string(" ")
+        printer.print_string("{")
+        if self.value_float:
+            printer.print_string("value_float")
+            printer.print_string(" = ")
+            printer.print_attribute(self.value_float)
+        elif self.value_floats:
+            pass
+        elif self.value_int:
+            printer.print_string("value_int")
+            printer.print_string(" = ")
+            printer.print_attribute(self.value_int)
+        elif self.value_ints:
+            printer.print_string("value_ints")
+            printer.print_string(" = ")
+            arr: list[int] = [value.value.data for value in self.value_ints]
+            printer.print(arr)
+        elif self.value_string:
+            pass
+        elif self.value_strings:
+            pass
+        printer.print_string("}")
+        printer.print_string(" : ")
+        printer.print_attribute(self.output.type)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        value_float = (
+            value_floats
+        ) = (
+            value_int
+        ) = value_ints = value_string = value_strings = value = output_type = None
+        if parser.parse_optional_punctuation("{"):
+            if parser.parse_optional_keyword("value_float"):
+                parser.parse_punctuation("=")
+                value_float = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_floats"):
+                parser.parse_punctuation("=")
+                value_floats = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_int"):
+                parser.parse_punctuation("=")
+                value_int = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_ints"):
+                parser.parse_punctuation("=")
+                value_ints = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_string"):
+                parser.parse_punctuation("=")
+                value_string = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_strings"):
+                parser.parse_punctuation("=")
+                value_strings = parser.parse_attribute()
+            parser.parse_punctuation("}")
+            parser.parse_punctuation(":")
+            output_type = parser.parse_type()
+        else:
+            value = parser.parse_attribute()
+            if isinstance(value, DenseIntOrFPElementsAttr):
+                output_type = value.type
```

Comment:
I'd expect this to work:
``` python
v = parser.parse_attribute()
if isinstance(v, DictionaryAttr):
  parser.parse_punctuation(':')
  t  = parser.parse_attribute()
else:
  # get the type somehow


constant = cls.build(attributes = {k: v}, result_types=(t,))
return constant
```

Reply:
But it all depends on how onnx-mlir does it

Reply:
it doesnt like it